### DISCUSSION TITLE
Gitlab Dependency Proxy in Dockerfile

### DISCUSSION THREAD

**Discussion by zenhighzer** [renovatebot/renovate#27917] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab CE

### Please tell us more about your question or problem

Hello there, 

for building Images and pushing them to my Registry I use the Gitlab Dependency Proxy. A Dockerfile typically looks like

```
FROM gitlab.my.site:443/mygroup/dependency_proxy/containers/ubuntu:jammy-20240212
RUN apt update && DEBIAN_FRONTEND=noninteractive apt install -y wget unzip curl python3 , etc. 
```

This way renovate can´t figure out if there are new updates

I little bit of google - I use the following renovate.json: 

```
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
      "config:recommended", "docker:disable"
      ],
    "regexManagers": [
      {
        "fileMatch": ["Dockerfile$"],
        "matchStrings": ["dependency_proxy\/containers\/(?<depName>.*):(?<currentValue>.*)\\n"],
        "datasourceTemplate": "docker"
      }
    ]
  }
```

Unfortunately this doesnt work... 

On the other side If I use an alpine-image for testing it works - Renovate finds new versions...: 

`FROM gitlab.my.site:443/mygroup/dependency_proxy/containers/alpine:3.19.1`

the regex seems to be fine: 

<img width="1217" alt="grafik" src="https://github.com/renovatebot/renovate/assets/65350020/6a6d937e-006d-4b4e-8bf5-1c68614a965f">

What am I missing? Help is much appreciated :) I am very new to renovate, just figuring out how this all works, please dont bite me :-D 

Thanks and Best Regards
Zen

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by mlec1** [comment]
I think your Ubuntu version is not supported yet, there is this [ticket](https://github.com/renovatebot/renovate/issues/27862) to support the `<ubuntu>-<date>` versioning

**Comment by zenhighzer** [comment]
Thank you @mlec1 :)

Is there any workaround? Does anyone got an idea?

**Comment by zenhighzer** [comment]
Oh wait a second, in https://github.com/renovate-reproductions/27806 there is written, that ubuntu-version-scheme works.... but not in my case? why? :)

**Reply by mlec1** [reply]
Ah yeah my bad, it is Debian versioning which is not working.

could you submit your Renovate log, that may be useful.

**Reply by zenhighzer** [reply]
@mlec1 No problem ;-) 

I will send the logs tomorrow, need a bed for now :-D Thank you for your help :-)

**Comment by zenhighzer** [comment, marked as answer]
Hi all, 

I think I found the error:  
There was something  missing in my regExManager:    "versioningTemplate": "ubuntu" 

This following config works, Renovate found a new ubuntu-version 
 :-) @mlec1  Thank you for the hint! ;-) 
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:recommended", "docker:disable"],
 "regexManagers": [
      {
        "fileMatch": ["Dockerfile$"],
        "matchStrings": ["dependency_proxy\/containers\/(?<depName>.*):(?<currentValue>.*)\\n"],
        "datasourceTemplate": "docker",
        "versioningTemplate": "ubuntu"
      }
    ] 
}
```

Have a nice day, Best Regards
Zen
